Fw/Buffer/Buffer.hpp

```diff
@@ -0,0 +1,172 @@
+// \brief  hpp file for Fw::Buffer definition
+//
+// \copyright
+// Copyright 2009-2020, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+#ifndef BUFFER_HPP_
+#define BUFFER_HPP_
+
+#include <Fw/Types/BasicTypes.hpp>
+#include <Fw/Types/Serializable.hpp>
+#if FW_SERIALIZABLE_TO_STRING
+    #include <Fw/Types/StringType.hpp>
+    #include <stdio.h> // snprintf
+    #ifdef BUILD_UT
+        #include <iostream>
+        #include <Fw/Types/EightyCharString.hpp>
+    #endif
+#endif
+
+namespace Fw {
+
+//! Buffer used for wrapping pointer to data for efficient transmission
+//!
+//! Fw::Buffer is a wrapper for a pointer to data. It allows for data to be passed around the system without a copy of
+//! the data itself. However, it comes with the expectation that the user maintain and protect this memory as it moves
+//! about the system until such a time as it is returned.
+//!
+//! Fw::Buffer is composed of several elements: a U8* pointer to the data, a U32 size of that data, and a U32 context
+//! describing the origin of that data, such that it may be freed at some later point. The default context of 0xFFFFFFFF
+//! should not be used for tracking purposes, as it represents a context-free buffer.
+//!
+//! Fw::Buffer also comes with functions to return a representation of the data as a SerializeBufferBase. These two
+//! functions allow easy access to the data as if it were a serialize or deserialize buffer. This can aid in writing and
+//! reading the wrapped data where as the standard serialize and deserialize methods treat the data as a pointer to
+//! prevent excessive copying.
+//!
+class Buffer : public Fw::Serializable {
+
+public:
+
+    enum {
+        SERIALIZED_SIZE = sizeof(U32) + sizeof(U32) + sizeof(U8*), //!< Size of Fw::Buffer when serialized
+        NO_CONTEXT = 0xFFFFFFFF //!< Value representing no context
+    };
+
+    //! Construct a buffer with no context nor data
+    //!
+    //! Constructs a buffer setting the context to the default no-context value of 0xffffffff. In addition, the size
+    //! and data pointers are zeroed-out.
+    Buffer();
+
+    //! Construct a buffer by copying data from a reference to anothe buffer
+    Buffer(const Buffer& src);
+
+    //! Construct a buffer to wrap the given data pointer of given size
+    //!
+    //! Wraps the given data pointer with given size in a buffer. The context by default is set to NO_CONTEXT but can
+    //! be set to specify a specific context.
+    //! \param data: data pointer to wrap
+    //! \param size: size of data located at data pointer
+    //! \param context: user-specified context to track creation. Default: no context
+    Buffer(U8* data, U32 size, U32 context=NO_CONTEXT);
+
+    //! Assignment operator to set given buffer from another
+    //!
+    Buffer& operator=(const Buffer& src);
```

Comment:
You may want to clarify in the comment that this is ***not*** a copy.

Reply:
Agreed.